docs/plugins/transform-react-inline-elements.md

```diff
@@ -10,14 +10,16 @@ Replaces the `React.createElement` function with one that is more optimized for
 
 <blockquote class="babel-callout babel-callout-info">
   <p>
-    Note that this requires the global polyfill due to React's use of ES6 <code>Symbol</code> when validating React Elements.
-    If <code>Symbol</code> is not present in the browser, your application will fail to render, as Babel will polyfill <code>Symbol</code>
-    but React will not have access to that polyfill.
+    Note that this might require the Symbol's global polyfill due to React's use of ES6 <code>Symbol</code> when validating React Elements.
```

Comment:
I think we can should add some additional context here. Maybe something like:

> Note that this requires `Symbol` to be available (or polyfilled) _globally_ since React uses them when validating elements.
>
> When using <code>transform-runtime</code>, polyfills (by default, including `Symbol`) are specifically scoped to not pollute globally, causing your application to fail when rendering as React will not have access to it.
>
> We recommend importing/requiring `babel-polyfill` in the entry point of your application and using `babel-preset-env` with the `useBuiltIns` option to only include the polyfills your targets need.

(feel free to edit/change/augment as much as you want)